compiler/resolution/functionResolution.cpp

```diff
@@ -6006,79 +5997,27 @@ static void resolveNewHandleInstantiatedInit(CallExpr* call) {
       resolveCall(call);
     }
 
-    resolveFunction(call->resolvedFunction());
+    resolveFunction(initFn);
 
     def->remove();
   }
 }
 
-static void resolveNewHandleGenericInitializer(CallExpr* call) {
-  SET_LINENO(call);
-
+// If the type was instantiated then determine the original generic type
+static AggregateType* resolveNewOriginalGenericType(CallExpr* call) {
   SymExpr*       typeExpr = resolveNewFindTypeExpr(call);
   AggregateType* at       = toAggregateType(resolveTypeAlias(typeExpr));
-  VarSymbol*     newTmp   = newTemp("new_temp", at);
-  DefExpr*       def      = new DefExpr(newTmp);
-  FnSymbol*      initFn   = NULL;
-
-  newTmp->addFlag(FLAG_DELAY_GENERIC_EXPANSION);
-
-  typeExpr->replace(new UnresolvedSymExpr("init"));
-
-  // Convert the PRIM_NEW to a normal call
-  call->primitive = NULL;
-  call->baseExpr  = call->get(1)->remove();
-
-  parent_insert_help(call, call->baseExpr);
-
-  fixupArgDefaultWhenRecordInit(at, call, newTmp);
-
-  if (isBlockStmt(call->parentExpr) == true) {
-    call->insertBefore(def);
-
-  } else {
-    Expr* parent = call->parentExpr;
+  AggregateType* retval   = at;
 
-    parent->insertBefore(def);
+  if (at->instantiatedFrom != NULL) {
```

Comment:
There's similar code in ResolutionCandidate.cpp (and perhaps elsewhere), so we should probably make this a method on AggregateType eventually. No need to do so for this PR.